memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala

```diff
@@ -172,8 +179,10 @@ final case class MutableHistogram(buckets: HistogramBuckets, values: Array[Doubl
   final def copy: Histogram = MutableHistogram(buckets, values.clone)
 
   /**
-   * Adds the values from another MutableHistogram having the same bucket schema.  If it does not, then
-   * an exception is thrown -- for now.  Modifies itself.
+   * Adds the values from another Histogram.
+   * If the other histogram has the same bucket scheme, then the values are just added per bucket.
+   * If the scheme is different, then an approximation is used so that the resulting histogram has
+   * an approximate sum of the individual distributions, with the original scheme.  Modifies itself.
    */
   final def add(other: HistogramWithBuckets): Unit =
     if (buckets == other.buckets) {
```

Comment:
Bucket comparison may be an O(n) operation depending on the type of Histogram buckets and that will be done very often. We should make this a constant time operation for fully enumerated buckets.

I am seeing that GeometricBuckets is the only type of histogram bucketing we have now. 
1. Are the optimizations holding good because we have only one implementation of this trait ?
2. At some point, we when we have enumerated buckets this code will be invalidated.

Reply:
It's probably impossible to make this constant time for all types of buckets.  We can possibly use some kind of hash for non-geometric buckets, which would make it mostly constant time.  

However, even if it was not constant time, adding buckets of the same scheme is still way faster than dealing with different schemes.